lms/static/js/commerce/track_ecommerce_events.js

```diff
@@ -0,0 +1,40 @@
+/**
+ *
+ * A library of helper functions to track ecommerce related events. 
+ *
+ */
+
+(function(define) {
+    'use strict';
+
+    define([
+        'jquery'
+    ], function($) {
+        var trackUpsellClick = function(linkName, optionalAttrs) {
```

Comment:
For the 8 simple links, this should work exactly as you have it. For the others, we won't know the page and link name at the time we write the link (since the code is rendered on multiple places). So trackUpsellClick will need to be able to handle having those two items missing. The good news is once we're in trackUpsellClick, we can infer all of the link names based on which page + category + code we called trackUpsellClick from (we pass this in another argument but that seems gross, maybe there's a simpler way?)

Reply:
I was thinking that wherever the link gets added, the logic to get the name would also live there, since the naming logic is specific to each page/button/link that we are adding the event to.

Reply:
The individual link code snippets that get pulled in don't have awareness of where they're being pulled in *from* so the info needed likely won't be in scope there. We'll likely need some outside javascript file to assist, but that can be a different upsell-tracking-logic file if we want to keep that outside of track_ecommerece_events

Reply:
Discussed offline, we want the business logic to live outside of the eventing module.